urlb44e9843ba_QuasimodoNZ_SWEN222_Game_Project_tgz-pJ8-Game_World_GameWorldLevelBuilderJ8/src/User_Interface/MediaPlayer.java

```diff
@@ -0,0 +1,78 @@
+package User_Interface;
+
+import java.io.File;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.Clip;
+
+/**
+ * A basic MediaPlayer that starts and stops songs.
+ * The different methods are called to play different songs
+ * depending on where in the game the player is:
+ *
+ * > welcomeScreen - played when the user is in the main menus
+ * > mainTheme - played when the player first joins the game
+ * > winningTheme - played when a player wins the game
+ *
+ * All credit for the music to HalcyonicFalconX at The Open Game
+ * Art Bundle, http://open.commonly.cc/
+ *
+ * @author Josh Brake, 300274198, brakejosh
+ */
+public class MediaPlayer {
+
+	private final static String mainTheme = "Sounds/An Adventure Awaits.wav";
+	private final static String welcomeScreen = "Sounds/The Frozen Journey.wav";
+	private final static String winningTheme = "Sounds/Rainbows and Unicorns.wav";
+	private static AudioInputStream ais;
+	private static Clip clip;
+
+	public MediaPlayer() { }
+
+	public static void playWelcome() {
+		try {
+			ais = AudioSystem
+					.getAudioInputStream(new File(welcomeScreen));
+			clip = AudioSystem.getClip();
+			clip.open(ais);
+			clip.loop(Clip.LOOP_CONTINUOUSLY);
+			clip.start();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 33:42, 46:55, 59:68.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.